hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -99,10 +103,32 @@ public HBaseIndex(HoodieWriteConfig config) {
   }
 
   private void init(HoodieWriteConfig config) {
-    multiPutBatchSize = config.getHbaseIndexGetBatchSize();
-    qpsFraction = config.getHbaseIndexQPSFraction();
-    maxQpsPerRegionServer = config.getHbaseIndexMaxQPSPerRegionServer();
-    putBatchSizeCalculator = new HbasePutBatchSizeCalculator();
+    this.multiPutBatchSize = config.getHbaseIndexGetBatchSize();
+    this.qpsFraction = config.getHbaseIndexQPSFraction();
+    this.maxQpsPerRegionServer = config.getHbaseIndexMaxQPSPerRegionServer();
+    this.putBatchSizeCalculator = new HbasePutBatchSizeCalculator();
+  }
+
+  public HBaseIndexQPSResourceAllocator createQPSResourceAllocator(HoodieWriteConfig config) {
+    try {
+      logger.info("createQPSResourceAllocator :" + config.getHBaseQPSResourceAllocatorClass());
+      final Class cl = Class.forName(config.getHBaseQPSResourceAllocatorClass());
+      final Constructor constructor = cl.getConstructor(HoodieWriteConfig.class);
+      final HBaseIndexQPSResourceAllocator resourceAllocator
+          = (HBaseIndexQPSResourceAllocator) constructor.newInstance(config);
+      return resourceAllocator;
+    } catch (ClassNotFoundException e) {
```

Comment:
Just catch Exception and pass it to the logger

Reply:
Also removed all other exceptions as it is handled at ReflectionUtils class.